src/hotspot/share/services/memoryManager.hpp

```diff
@@ -57,11 +57,14 @@ class MemoryManager : public CHeapObj<mtInternal> {
   const char* _name;
 
 protected:
-  volatile OopHandle _memory_mgr_obj;
+  OopHandle _memory_mgr_obj;
+  volatile bool _memory_mgr_obj_initialized;
 
-public:
   MemoryManager(const char* name);
 
+public:
+  virtual ~MemoryManager() = default; // FIXME
```

Comment:
Was this added because we are deleting these objects through the base class pointer? Or just in case?

Regardless what is the `FIXME` comment for?

Reply:
Oops, that was a piece of another change that I started but abandoned for now.
Removed.